src/cp/slipharden.h

```diff
@@ -20,50 +22,50 @@ namespace neml {
 class SlipRule; // Why would we need a forward declaration?
```

Comment:
this 'class' does not have the NEML_EXPORT ?

Reply:
That's a forward declaration, it doesn't need the export symbol.